src/Element/Collection.php

```diff
@@ -162,11 +161,13 @@ public function allowObjectBinding($object)
      */
     public function setObject($object)
     {
-        if (! is_array($object) && ! $object instanceof Traversable) {
+        if ($object instanceof Traversable) {
+            $object = iterator_to_array($object);
```

Comment:
```suggestion
            $object = ArrayUtils::iteratorToArray($object);
```

Reply:
Nope, as `ArrayUtils::iteratorToArray` is recursive by default and at this point we don't want to replace it recursively - see tests.

Of course you could say to set 2nd param to false - but then it will just use `iterator_to_array` internally with additional redundant checks.